docs/ops/memory/mem_setup.md

```diff
@@ -0,0 +1,251 @@
+to specify explicitly both [task heap](#task-operator-heap-memory) and [managed memory](#managed-memory).
+It gives more control over the available JVM heap to Flink’s tasks and its [managed memory](#managed-memory).
+
+The rest of the memory components will be adjusted automatically, based on default values or additionally configured options.
+[Here](#detailed-memory-model) are more details about the other memory components.
+
+<strong>Note:</strong> If you have configured the task heap and managed memory explicitly, it is recommended to set neither
+*total process memory* nor *total Flink memory*. Otherwise, it may easily lead to memory configuration conflicts.
+
+### Task (Operator) Heap Memory
+
+If you want to guarantee that a certain amount of JVM heap is available for your user code, you can set the *task heap memory*
+explicitly ([taskmanager.memory.task.heap.size](../config.html#taskmanager-memory-task-heap-size)).
+It will be added to the JVM heap size and will be dedicated to Flink’s operators running the user code.
+
+### Managed Memory
+
+*Managed memory* is managed by Flink and is allocated as native memory (off-heap). The following workloads use *managed memory*:
+* Streaming jobs can use it for [RocksDB state backend](../state/state_backends.html#the-rocksdbstatebackend).
+* [Batch jobs](../../dev/batch) can use it for sorting, hash tables, caching of intermediate results.
+
+The size of *managed memory* can be
+* either configured explicitly via [taskmanager.memory.managed.size](../config.html#taskmanager-memory-managed-size)
+* or computed as a fraction of *total Flink memory* via [taskmanager.memory.managed.fraction](../config.html#taskmanager-memory-managed-fraction)
+
+*size* will override *fraction*, if both are set.
+If neither *size* nor *fraction* is explicitly configured, the [default fraction](../config.html#taskmanager-memory-managed-fraction) will be used.
+
+See also [how to configure memory for state backends](mem_tuning.html#configure-memory-for-state-backends) and [batch jobs](mem_tuning.html#configure-memory-for-batch-jobs).
+
+## Configure Off-Heap Memory (direct or native)
+
+The off-heap memory which is allocated by user code should be accounted for in *task off-heap memory*
+([taskmanager.memory.task.off-heap.size](../config.html#taskmanager-memory-task-off-heap-size)).
+
+<strong>Note:</strong> You can also increase the [framework off-heap memory](#framework-memory). This option is advanced
+and only recommended to be changed if you are sure that the Flink framework needs more memory.
+
+Flink includes the *framework off-heap memory* and *task off-heap memory* into the *direct memory* limit of the JVM,
+see also [JVM parameters](#jvm-parameters).
+
+<strong>Note:</strong> Although, native non-direct memory usage can be accounted for as a part of the
+*framework off-heap memory* or *task off-heap memory*, it means that the JVM's *direct memory* limit will be higher.
```

Comment:
Would this be easier to understand?
`Although, native non-direct memory usage can be accounted for as a part of the
*framework off-heap memory* or *task off-heap memory*, it will result in a higher JVM's *direct memory* limit in this case.`